packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxis.ts

```diff
@@ -125,6 +163,85 @@ export const useChartCartesianAxis: ChartPlugin<UseChartCartesianAxisSignature<a
     isInteractionEnabled,
   ]);
 
+  React.useEffect(() => {
+    const subscribeModifications = (
+      newStoreState: ChartState<
+        [UseChartCartesianAxisSignature<any>],
+        [UseChartInteractionSignature]
+      >,
+      prevStoreState: ChartState<
+        [UseChartCartesianAxisSignature<any>],
+        [UseChartInteractionSignature]
+      >,
+    ) => {
+      if (!onAxisInteraction) {
+        return;
+      }
+
+      const newXAxisIds = selectorChartRawXAxis(newStoreState)?.map((axis) => axis.id) ?? [];
+      const newYAxisIds = selectorChartRawYAxis(newStoreState)?.map((axis) => axis.id) ?? [];
+
+      const newAxisPointerInteraction: AxisItemIdentifier[] = [];
+      let isSame = true;
+
+      for (const axisId of newXAxisIds) {
+        const prevAxisIndex = selectorChartsInteractionXAxisIndex(prevStoreState, axisId);
+        const nextAxisIndex = selectorChartsInteractionXAxisIndex(newStoreState, axisId);
```

Comment:
Use selectors to know the dataIndex corresponding the the pointer position in the previous and next state